tests/kafka/pykafka/test_protocol.py

```diff
@@ -177,5 +177,56 @@ def test_response(self):
         self.assertEqual(resp.topics['test'], {0: [2]})
 
 
+class TestOffsetCommitFetchAPI(unittest.TestCase):
+    def test_consumer_metadata_request(self):
+        req = protocol.ConsumerMetadataRequest('test')
+        msg = req.get_bytes()
+        self.assertEqual(
+            msg,
+            bytearray(b'\x00\x00\x00\x1b\x00\n\x00\x00\x00\x00\x00\x00\x00\x07pykafka\x00\x00\x00\x00\x00\x04test')
+        )
+
+    def test_consumer_metadata_response(self):
+        response = protocol.ConsumerMetadataResponse(
+            buffer('\x00\x00\x00\x00\x00\x00\x00\remmett-debian\x00\x00#\x84')
+        )
+        self.assertEqual(response.coordinator_id, 0)
+        self.assertEqual(response.coordinator_host, 'emmett-debian')
+        self.assertEqual(response.coordinator_port, 9092)
+
+    def test_offset_commit_request(self):
+        preq = protocol.PartitionOffsetCommitRequest('test', 0, 68, 1426632066,
+                                                     'testmetadata')
+        req = protocol.OffsetCommitRequest('test', partition_requests=[preq, ])
+        msg = req.get_bytes()
+        self.assertEqual(
+            msg,
+            bytearray(b'\x00\x00\x00G\x00\x08\x00\x00\x00\x00\x00\x00\x00\x07pykafka\x00\x04test\x00\x00\x00\x01\x00\x04test\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00D\x00\x00\x00\x00U\x08\xad\x82\x00\x0ctestmetadata')
+        )
+
+    def test_offset_commit_response(self):
+        response = protocol.OffsetCommitResponse(
+            buffer('\x00\x00\x00\x01\x00\x0cemmett.dummy\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00')
+        )
+        self.assertEqual(response.topics['emmett.dummy'][0], 0)
+
+    def test_offset_fetch_request(self):
+        preq = protocol.PartitionOffsetFetchRequest('testtopic', 0)
+        req = protocol.OffsetFetchRequest('test', partition_requests=[preq, ])
+        msg = req.get_bytes()
+        self.assertEqual(
+            msg,
+            bytearray(b'\x00\x00\x00.\x00\t\x00\x00\x00\x00\x00\x00\x00\x07pykafka\x00\x04test\x00\x00\x00\x01\x00\ttesttopic\x00\x00\x00\x01\x00\x00\x00\x00')
+        )
+
+    def test_offset_fetch_response(self):
+        # TODO - this is what I get back from kafka at the moment, clearly it's wrong
+        response = protocol.OffsetFetchResponse(
+            buffer('\x00\x00\x00\x01\x00\x0cemmett.dummy\x00\x00\x00\x01\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x01\x00\x00\x00\x00')
+        )
+        self.assertEqual(response.topics['emmett.dummy'][0].metadata, '')
```

Comment:
@kbourgoin One thing I noticed when testing is that I can never get an OffsetFetchResponse to return anything other than `''` in the `metadata` field.